### ISSUE TITLE
Labeling method

### ISSUE THREAD

**Issue by auesro** [AlbertPun/TRAILMAP#2]
Dear @AlbertPun,

I find the major hurdle here to start using the program is to label our own samples. If you could provide some script (as undocumented or basic as it might be) to perform that task, it would be a great help.

Several doubts:
-How many training samples does the network requires if training from scratch?
-How many training samples does the network requires for transfer learning?
-Do you define a training sample as each 64x64x64 px cube with 1-2 labeled slices?

Thanks a lot,

Best,

Augusto

**Comment by albert597** [comment]
Hello,

You only need to provide your own data if the current network does not work to a satisfactory level on your own data, which may deviate from the data we trained on. Have you tried running the current network on your data with the python3 segment_brain_batch.py BRAIN_FOLDER?

If you still find certain issues with a certain type of axon, you can label your own data of the axons or artifacts the network is doing poorly on. We found ImageJ's Segmentation editor (https://imagej.net/Segmentation_Editor) to be simple and easy to use.

Answer to several doubts:

> Do you define a training sample as each 64x64x64 px cube with 1-2 labeled slices?

I define an ORIGINAL training example/sample as the ~128 length cubes while I call the ~64 length cubes training examples/sample. 

> How many training samples does the network requires if training from scratch?

If you would like to train from scratch, you will need around n = 50-100 original training samples (~128 length cubes). From these large cubes, you run python3 prepare_data.py "generate_training_set" to generate 100 * n small ~64 length cubes cropped from the larger cubes (100 training samples per an original training sample). If possible, I recommend you to not train from scratch since these original training examples would take a significant amount of effort to label.

> How many training samples does the network requires for transfer learning?

Depending on how different your axons are from our training set, you will probably need around 4-10 original training examples (400-1000 training samples). You may want to include some artifact/blank chunks (examples where there are no axons) to make sure the network does not overfit to the axons. 

Please let me know if you have any other questions!

**Comment by auesro** [comment]
Hi Albert,

Thanks a lot for your answers. 

No, I havent started yet, I am planning how to best process the brains. Of course, I will try first your model without further training and see how it performs.

Will most likely come back to you!

Cheers
